### PR TITLE
feat: SE-106 Added select file type component

### PR DESCRIPTION
**Pull request by Ldarnb**

### CODE REVIEW

File: src/components/SelectFileType/SelectFileType.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    })
+
+
+    const emit = defineEmits(['select'])
+    
+    const tidyExtension = (e) => {
+        var extension = e;
+        extension = extension.toUpperCase().trim().replace('.', '');
+        return extension;
+    }
```
Comment (danielstorey) [inline src/components/SelectFileType/SelectFileType.vue:80]: Line 77 doesn't really accomplish anything and generally you don't really need to declare a variable if you are instantly returning it. var is no longer used really. You can use `let` if you want to redeclare a variable but in this case you don't even need to do that.

 ```suggestion
    const tidyExtension = (extension) => {
        return extension.toUpperCase().trim().replace('.', '');
    }
```

File: src/components/SelectFileType/SelectFileType.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
+<template>
+
+    <div 
+        @click="selectFile()"
+        class="inline-flex items-center border border-steel text-center h-24 w-24 rounded-md border-2 relative cursor-pointer overflow-hidden justify-center"
+        :class="props.selected ? 'bg-indigo fill-white' : ' bg-white fill-gray-500'">
```
Comment (danielstorey) [inline src/components/SelectFileType/SelectFileType.vue:6]: You don't need to use `props` in the template, only the script. It works, but not necessary. Do this for other instances below

```suggestion
        :class="selected ? 'bg-indigo fill-white' : ' bg-white fill-gray-500'">
```

**Review by danielstorey — COMMENTED** [review]
Looking good. A few minor points below.

File: src/components/SelectFileType/SelectFileType.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    const props = defineProps({
+        selected: { 
+            Type: Boolean, 
+            default: false 
+        },
+        label: {
+            Type: String,
+            default: 'Label'
+        },
+        extension: {
```
Comment (danielstorey) [inline src/components/SelectFileType/SelectFileType.vue:67]: maybe refactor to `ext` as it's shorter and a well used and understood abbreviation

File: src/components/SelectFileType/SelectFileType.vue [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <CSelectFileTypeItem
```
Comment (danielstorey) [inline src/components/SelectFileType/SelectFileType.vue:2]: I'd consider wrapping this in a `<div>` to control layout/spacing

File: src/components/SelectFileType/SelectFileType.vue [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  v-for="type in types"
+  :selected="type.ext === currentFileType"
+  :label="type.label"
+  :ext="type.ext"
+  @click="currentFileType = type.ext"
+  >
```
Comment (danielstorey) [inline src/components/SelectFileType/SelectFileType.vue:8]: Use self closing tag for empty elements

**Review by mattbarton44 — APPROVED** [review]

**Review by danielstorey — APPROVED** [review]

**Pull request merged**